### ISSUE TITLE
ImageDataBunch.from_csv does not register test dataset

### ISSUE THREAD

**Issue by keitakurita** [fastai/fastai#1072]
First of all, thank you so much for this amazing library! I have looked through the forums but did not find anyone with a similar issue so am filing this issue on GH.

**Describe the bug**
When constructing a databunch using ImageDataBunch.from_csv with a test set, the test dataset is not registered with the resulting DataBunch.

**To Reproduce**
```python
from fastai import *
from fastai.vision import *

# create sample data
pd.DataFrame({"fn": ["a.jpg", "b.jpg", "c.jpg"], "lbl": [0, 1, 2]}).to_csv("tst_tmp.csv", index=False)
Path("fastai_dbg/train").mkdir(parents=True, exist_ok=True)
Path("fastai_dbg/test").mkdir(parents=True, exist_ok=True)
import scipy.misc
scipy.misc.imsave('fastai_dbg/train/a.jpg', np.random.rand(64, 64))
scipy.misc.imsave('fastai_dbg/train/b.jpg', np.random.rand(64, 64))
scipy.misc.imsave('fastai_dbg/train/c.jpg', np.random.rand(64, 64))
scipy.misc.imsave('fastai_dbg/test/d.jpg', np.random.rand(64, 64))
scipy.misc.imsave('fastai_dbg/test/e.jpg', np.random.rand(64, 64))

# create databunch
databunch = ImageDataBunch.from_csv(".", folder="fastai_dbg", csv_labels="tst_tmp.csv",
                                    valid_pct=0.1, test="test", fn_col=0, label_col=1)

# throws error: You didn't specify a test set for this DataBunch
databunch.test_ds
```

**Expected behavior**
The test dataset/dataloader should be added.

**Additional context**
I suspect that this has to do with how bool(dl: DataLoader) is evaluated. The reason the data loader is not being loaded is because for some reason, in the line 
```python
self.test_dl  = DeviceDataLoader(test_dl, self.device, self.tfms, collate_fn) if test_dl else None
```
in the constructor of DataBunch (basic_data.py, line 103), `if test_dl` is evaluated to False.
This may be a problem of the PyTorch version I have installed, but unless this clause is meant to guard agains some malformed datasets, I think it is better to explicitly write
```python
self.test_dl  = DeviceDataLoader(test_dl, self.device, self.tfms, collate_fn) if test_dl is not None else None
```
to avoid this dependency with PyTorch. 

If this is an issue that seems worth fixing I'm perfectly willing to file a quick PR for it.

**Comment by sgugger** [comment]
Good catch! Yes a PR would be appreciated as in multiple instances, is not None seems to work better.

**Comment by sgugger** [comment]
Also, if you can add a quick test that fails if the test dl doesn't load properly when your fix isn't there, that would be the cherry on the top!
